L1Trigger/TrackTrigger/interface/TTClusterAlgorithm.h

```diff
@@ -0,0 +1,78 @@
+/*! \class   TTClusterAlgorithm
+ *  \brief   Base class for any algorithm to be used
+ *           in TTClusterBuilder
+ *  \details After moving from SimDataFormats to DataFormats,
+ *           the template structure of the class was maintained
+ *           in order to accomodate any types other than PixelDigis
+ *           in case there is such a need in the future.
+ *
+ *  \author Andrew W. Rose
+ *  \author Nicola Pozzobon
+ *  \date   2013, Jul 12
+ *
+ */
+
+#ifndef L1_TRACK_TRIGGER_CLUSTER_ALGO_BASE_H
+#define L1_TRACK_TRIGGER_CLUSTER_ALGO_BASE_H
+
+#include <sstream>
+#include <map>
+#include <string>
+#include "classNameFinder.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+
+template< typename T >
+class TTClusterAlgorithm
+{
+  protected:
+    /// Data members
+    std::string                  className_;
```

Comment:
This is only used for printout... if this is a problem, it could be removed all together, I guess.